### PR TITLE
Shot Generator 2.0: Volume Object

### PR DESCRIPTION
**Pull request by tvvik**
- added volumes.json preset file containing the current default volume presets 
- added 'Volumetric' class that loads a set of images and displays them on layers
- added 'VolumePresetsEditor' class that handles selecting and creating new volumes

known issues where i'd like @audionerd input:

- [x] couldn't set up the NumberSlider for a integer step of 1, will comment on exact spot
- [x] volumes.json doesn't get saved between sessions, so if you create a new effect and save a board with it, it will ~~throw errors~~ alert and discard the volume when you try to load it again because it doesn't find the effect

- [ ] maybe save images for created effects on server? 

- [x] copy user selected image files to project folder

known issue: https://github.com/wonderunit/storyboarder/issues/1606

### CODE REVIEW

File: src/js/shot-generator/Editor.js [hunk 1]
```diff
@@ -1809,13 +1909,44 @@ const InspectedElement = ({ sceneObject, models, updateObject, selectedBone, mac
...
 
+      sceneObject.type == 'volume' && [        
+        [
+          'div.column',
+          // [VolumePresetsEditor2, { sceneObject }],
+          [VolumePresetsEditor, { sceneObject }],
+          [NumberSlider, { label: 'width', value: sceneObject.width, min: 0.1, max: 25, onSetValue: createOnSetValue(sceneObject.id, 'width') } ],
+          [NumberSlider, { label: 'height', value: sceneObject.height, min: -25, max: 25, onSetValue: createOnSetValue(sceneObject.id, 'height') } ],
+          [NumberSlider, { label: 'depth', value: sceneObject.depth, min: 0.1, max: 25, onSetValue: createOnSetValue(sceneObject.id, 'depth') } ], 
+          [NumberSlider, {
```
Comment (tvvik) [inline src/js/shot-generator/Editor.js]: This is the one i didn't manage to set up, so right now i'm just converting the values i get to int

File: .gitignore [hunk 2]
```diff
@@ -41,3 +41,5 @@ dist/
 .vscode
 
 .DS_Store
+.editorconfig
```
Comment (audionerd) [inline .gitignore]: Let's not .gitignore these globally. If they bother you, you can ignore locally (e.g.: `.git/info/exclude`). We'll add them to the repo soon.

File: src/js/shot-generator/Editor.js [hunk 3]
```diff
@@ -99,11 +101,12 @@ const NumberSliderTransform = {
     if (value > 180) { return value - 360 }
     if (value < -180) { return value + 360 }
     return value
-  }
+  },
+  intNumber: value => {return parseInt(value)}
```
Comment (audionerd) [inline src/js/shot-generator/Editor.js]: Is this used?

File: src/js/shared/reducers/shot-generator.js [hunk 4]
```diff
@@ -691,6 +695,26 @@ module.exports = {
...
 
+          if (action.payload.numberOfLayers != null) {
+            draft.sceneObjects[action.payload.id].numberOfLayers = action.payload.numberOfLayers
+          }
+
+          if (action.payload.distanceBetweenLayers != null) {
+            draft.sceneObjects[action.payload.id].distanceBetweenLayers = action.payload.distanceBetweenLayers
+          }
+
+          if (action.payload.effect != null) {
```
Comment (audionerd) [inline src/js/shared/reducers/shot-generator.js]: What is `effect`? Should it be renamed `volumePresetId`?

Comment (tvvik) [reply src/js/shared/reducers/shot-generator.js]: changed, was left from when i started adding effects and they just had names

File: .gitignore (hunk 2 above)
Comment (tvvik) [reply .gitignore]: done

**Pull request merged**